src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -202,20 +218,59 @@ class RealHyperShard(
         }
         return tests
     }
-}
 
-fun main(args: Array<String>) {
-    checkUsage(args)
-    val annotationName = args[0]
-    val dirs = args.sliceArray(1 until args.size)
-    val processor = RealHyperShard(annotationName, dirs)
-    processor.gatherTests().stream().forEach(::println)
+    /**
+     * Returns true if we found the class annotation or if's meant to be empty
+     */
+    private fun shouldProcessTestMethods(
+        ktClass: KtClass,
+        annotationValue: ClassAnnotationValue
+    ): Boolean {
+        return when (annotationValue) {
+            is ClassAnnotationValue.Present -> {
+                var found = false
+                for (annotationEntry in ktClass.annotationEntries) {
+                    if (annotationEntry.shortName.toString() == annotationValue.annotationName) {
+                        found = true
```

Comment:
why not return true from here and false outside the loop, that way you won't need the flag

Reply:
👍